MSAL/src/MSALPublicClientApplication.m

```diff
@@ -964,7 +964,7 @@ - (void)acquireTokenSilentForScopes:(NSArray<NSString *> *)scopes
         }
 
         NSError *resultError = nil;
-        MSALResult *msalResult = [MSALResult resultWithTokenResult:result error:&resultError];
+        MSALResult *msalResult = [MSALResult resultWithTokenResult:result tokenCache:_tokenCache error:&resultError];
```

Comment:
why do you need to pass token cache to result?

Reply:
Because `MSIDAccount` in `MSIDTokenResult` is not sufficient to construct all tenant profiles, it is only for one tenant. Cache lookup is needed to find all tenant profiles :)